src/vs/workbench/services/editor/test/browser/editorService.test.ts

```diff
@@ -178,6 +178,723 @@ suite('EditorService', () => {
 		didCloseEditorListener.dispose();
 	});
 
+	test('openEditor - override handling', () => {
+		return testOpenEditors(false);
+	});
+
+	test('openEditors - override handling', () => {
+		return testOpenEditors(true);
+	});
+
+	async function testOpenEditors(useOpenEditors: boolean) {
+		disposables.add(registerTestFileEditor());
+
+		const [part, service, accessor] = await createEditorService();
+
+		let rootGroup = part.activeGroup;
+
+		let editorFactoryCalled = 0;
+		let untitledEditorFactoryCalled = 0;
+		let diffEditorFactoryCalled = 0;
+
+		let lastEditorFactoryEditor: IResourceEditorInput | undefined = undefined;
+		let lastUntitledEditorFactoryEditor: IUntitledTextResourceEditorInput | undefined = undefined;
+		let lastDiffEditorFactoryEditor: IResourceDiffEditorInput | undefined = undefined;
+
+		disposables.add(accessor.editorOverrideService.registerEditor(
+			'*.editor-service-override-tests',
+			{ id: 'editorServiceOverrideTests', label: 'Label', priority: RegisteredEditorPriority.exclusive },
+			{ canHandleDiff: true },
```

Comment:
Not super related to this PR but what yours thoughts on abandoning `canHandledDiff` since we don't have a `canHandleUntitled` as we just check for the factory function.

Reply:
Yeah awesome idea, please do that 👍